src/System.Net.Sockets/tests/FunctionalTests/SendFile.cs

```diff
@@ -0,0 +1,211 @@
+                random.NextBytes(preBuffer);
+                checksum.Add(preBuffer, 0, preBuffer.Length);
+            }
+
+            byte[] fileBuffer = new byte[fileSize];
+            random.NextBytes(fileBuffer);
+
+            string path = Path.GetTempFileName();
+            File.WriteAllBytes(path, fileBuffer);
+
+            checksum.Add(fileBuffer, 0, fileBuffer.Length);
+
+            postBuffer = null;
+            if (sendPreAndPostBuffers)
+            {
+                postBuffer = new byte[256];
+                random.NextBytes(postBuffer);
+                checksum.Add(postBuffer, 0, postBuffer.Length);
+            }
+
+            return path;
+        }
+
+        [OuterLoop] // TODO: Issue #11345
+        [Theory]
+        [MemberData(nameof(SendFile_MemberData))]
+        private void SendFile_Synchronous(IPAddress listenAt, bool sendPreAndPostBuffers)
+        {
+            const int BytesToSend = 123456;
+            const int ListenBacklog = 1;
+            const int LingerTime = 10;
+            const int TestTimeout = 30000;
+
+            // Create file to send
+            byte[] preBuffer;
+            byte[] postBuffer;
+            Fletcher32 sentChecksum;
+            string filename = CreateFileToSend(BytesToSend, sendPreAndPostBuffers, out preBuffer, out postBuffer, out sentChecksum);
+
+            // Start server
+            var server = new Socket(listenAt.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+            server.BindToAnonymousPort(listenAt);
+
+            server.Listen(ListenBacklog);
+
+            int bytesReceived = 0;
+            var receivedChecksum = new Fletcher32();
+            var serverThread = new Thread(() =>
+            {
+                using (server)
+                {
+                    Socket remote = server.Accept();
+                    Assert.NotNull(remote);
+
+                    using (remote)
+                    {
+                        var recvBuffer = new byte[256];
+                        for (;;)
+                        {
+                            int received = remote.Receive(recvBuffer, 0, recvBuffer.Length, SocketFlags.None);
+                            if (received == 0)
+                            {
+                                break;
+                            }
+
+                            bytesReceived += received;
+                            receivedChecksum.Add(recvBuffer, 0, received);
+                        }
+                    }
+                }
+            });
+            serverThread.Start();
+
+            // Run client
+            EndPoint clientEndpoint = server.LocalEndPoint;
+            var client = new Socket(clientEndpoint.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+            client.Connect(clientEndpoint);
+
+            using (client)
+            {
+                client.SendFile(filename, preBuffer, postBuffer, TransmitFileOptions.UseDefaultWorkerThread);
```

Comment:
It'd probably be good to validate some other scenarios:
- Sending a zero-length file, a very small file, a very large file (several megabytes), etc.
- Sending a FileStream for something other than a file on disk, e.g. a pipe (I'm not sure if that's supposed to work or not)
- Error conditions, like when the specified file doesn't exist or can't be opened due to lack of permissions